### PR TITLE
A fix of adding new sub-nodes

### PR DESCRIPTION
**Pull request by MrCry0**
Previous implementation of add_descriptor_node() does not support
      a case if a middle-way sub-node exists. This leads to panic if
      add 2 different sub-nodes in the same path. Fix such case handling
      a return code from fdt_add_subnode().
      
Fixes: 81cf62731 ("[FIO toup] arm: boot: reserve devices for exclusive access")

### CODE REVIEW

File: core/arch/arm/kernel/boot.c [hunk 1]
```diff
@@ -1218,9 +1218,15 @@ static int add_descriptor_node(struct dt_descriptor *dt, const char *driver)
...
+		int parentoffs = offs;
+		offs = fdt_add_subnode(dt->blob, parentoffs, item[j]);
+		if (offs >= 0)
+			continue;
+		if (offs == -FDT_ERR_EXISTS)
+			offs = fdt_subnode_offset(dt->blob, parentoffs,
+						  item[j]);
 		if (offs < 0)
-			return -1;
+			return  -1;
```
Comment (ricardosalveti) [inline core/arch/arm/kernel/boot.c]: This space is removed on the commit that follows, but mind removing it as part of this patch instead?

Comment (MrCry0) [reply core/arch/arm/kernel/boot.c]: @ricardosalveti thanks, fixed.

**Review by quaresmajose — APPROVED** [review]
LGTM

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**